File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/typescript/008_prime_numbers/solution.ts
import { readFileSync } from 'fs';

// Read command line arguments
// process.argv[0] is the node/ts executable
// process.argv[1] is the script name
// process.argv[2] is the first argument provided by the user
const inputArg = process.argv[2];

if (!inputArg) {
  // No argument provided, exit silently or handle as per requirements (print nothing)
  process.exit(0);
}

const N = parseInt(inputArg, 10);

if (isNaN(N) || N < 2) {
  // If N is not a valid number or less than 2, there are no primes.
  process.exit(0);
}

// Sieve of Eratosthenes to find primes up to N
const isPrime = new Array<boolean>(N + 1).fill(true);
isPrime[0] = false;
isPrime[1] = false;

for (let i = 2; i * i <= N; i++) {
  if (isPrime[i]) {
    for (let j = i * i; j <= N; j += i) {
      isPrime[j] = false;
    }
  }
}

// Collect and print primes
const primes: number[] = [];
for (let i = 2; i <= N; i++) {
  if (isPrime[i]) {
    primes.push(i);
  }
}

if (primes.length > 0) {
  console.log(primes.join('\n'));
}

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) < 2 {
		return
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		// Invalid input, do nothing as per requirements (or could error, but spec implies valid int)
		return
	}

	if n < 2 {
		return
	}

	// Sieve of Eratosthenes to find primes up to n
	isPrime := make([]bool, n+1)
	for i := 2; i <= n; i++ {
		isPrime[i] = true
	}

	for i := 2; i*i <= n; i++ {
		if isPrime[i] {
			for j := i * i; j <= n; j += i {
				isPrime[j] = false
			}
		}
	}

	for i := 2; i <= n; i++ {
		if isPrime[i] {
			fmt.Println(i)
		}
	}
}